contracts/ERC721CM.sol

```diff
@@ -0,0 +1,649 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/common/ERC2981.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "contracts/creator-token-standards/erc721c/v2/ERC721AC.sol";
+import "./IERC721M.sol";
+
+/**
+ * @title ERC721CM
+ *
+ * @dev ERC721A and ERC721C subclass with MagicEden launchpad features including
+ *  - multiple minting stages with time-based auto stage switch
+ *  - global and stage wallet-level minting limit
+ *  - whitelist using merkle tree
+ *  - crossmint support
+ *  - anti-botting
+ */
+contract ERC721CM is IERC721M, ERC721AC, Ownable, ReentrancyGuard {
```

Comment:
why don't we just inherit from ERC721M?

Reply:
Because ERC721M -> ERC721A, while ERC721C is also depending on ERC721. It's much cleaner to just extend ERC721AC then add our customization